Sources/ViewModels/ALKConversationViewModel.swift

```diff
@@ -1677,8 +1677,27 @@ open class ALKConversationViewModel: NSObject, Localizable {
         alMessage.source = Int16(AL_SOURCE_IOS)
         alMessage.conversationId = conversationId
         alMessage.groupId = channelKey
+        setMetaDataForMessage(alMessage: alMessage)
         return alMessage
     }
+    
+    private func setMetaDataForMessage(alMessage : ALMessage){
+        do {
+            var languageDict : [String:String] = [:]
+            let languageCode = NSLocale.preferredLanguages.first?.prefix(2)
+            if let languageCodeString = languageCode.map(String.init) {
+                languageDict["kmUserLocale"] = languageCodeString
+            }
+            let messageInfoData = try JSONSerialization.data(withJSONObject: languageDict, options: .prettyPrinted)
+            let messageInfoString = String(data: messageInfoData, encoding: .utf8) ?? ""
+            if(alMessage.metadata == nil){
+                alMessage.metadata = NSMutableDictionary()
+            }
```

Comment:
optimise this ..try to wrap it in single if let

Reply:
done